### PR TITLE
B2CA-1522: Add sender field in transaction

### PR DESCRIPTION
**Pull request by cedelavergne-ledger**
In addition to the `To` field, add the `From` to ease the transaction addresses reviews

## Description

Add new field in the structure
Add new field in the ValuePair review information
Update tests and screenshots

## Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [X] Tests
- [ ] Documentation
- [ ] Other (for changes that might not fit in any category)

### CODE REVIEW

File: src_features/signTx/logic_signTx.c [hunk 1]
```diff
@@ -492,6 +493,20 @@ __attribute__((noinline)) static bool finalize_parsing_helper(bool direct, bool
             strlcpy(strings.common.fullAmount, displayBuffer, sizeof(strings.common.fullAmount));
         }
         PRINTF("Amount displayed: %s\n", strings.common.fullAmount);
+
+        if (G_called_from_swap) {
+            // Transaction parameters are managed by the Exchange caller app!
+            explicit_bzero(strings.common.fromAddress, sizeof(strings.common.fromAddress));
+        } else {
+            // Format the from address in a temporary buffer, if in swap case compare it with
```
Comment (fbeutin-ledger) [inline src_features/signTx/logic_signTx.c:501]: not in swap case

File: tests/ragger/test_domain_name.py [hunk 2]
```diff
@@ -30,21 +30,41 @@ def verbose_fixture(request) -> bool:
...
-def common(firmware: Firmware, app_client: EthAppClient) -> int:
+def common_challenge(firmware: Firmware, app_client: EthAppClient) -> int:
 
     if firmware.device == "nanos":
         pytest.skip("Not supported on LNS")
     challenge = app_client.get_challenge()
     return ResponseParser.challenge(challenge.data)
 
 
+def common_nav(firmware: Firmware,
```
Comment (fbeutin-ledger) [inline tests/ragger/test_domain_name.py:41]: If you want you can use the simple navigation helper from ragger. It does just that

Comment (fbeutin-ledger) [reply tests/ragger/test_domain_name.py:41]: https://github.com/LedgerHQ/app-boilerplate/pull/119/files#diff-833eec1ba57c6e4d41959eb8ae23f6b8c0709eef144acf1445937b0342753dd6

Comment (cedelavergne-ledger) [reply tests/ragger/test_domain_name.py:41]: Yes, it is already the case with the up-to-date `develop` branch.
This commit has been skipped now, after the rebase.

File: src_features/signTx/logic_signTx.c (hunk 1 above)
Comment (cedelavergne-ledger) [reply src_features/signTx/logic_signTx.c:501]: Comment fixed

**Review by apaillier-ledger — CHANGES_REQUESTED** [review]

File: src/shared_context.h [hunk 3]
```diff
@@ -127,6 +127,7 @@ typedef enum {
 #define NETWORK_STRING_MAX_SIZE 19
 
 typedef struct txStringProperties_s {
+    char fromAddress[43];
     char fullAddress[43];
```
[the remark this answers is not in the document]
Comment (cedelavergne-ledger) [reply src/shared_context.h:131]: ok, done.

**Review by apaillier-ledger — APPROVED** [review]

**Review by apaillier-ledger — APPROVED** [review]

**Pull request merged**